### PR TITLE
[SYCL] Add `TIMESTEP_EMBEDDING` OP

### PR DESCRIPTION
**Pull request by zhentaoyu**
- [x] I have read the [contributing guidelines](https://github.com/ggerganov/llama.cpp/blob/master/CONTRIBUTING.md)
- Self-reported review complexity:
  - [x] Low
  - [ ] Medium
  - [ ] High

Add SYCL `TIMESTEP_EMBEDDING` operator by following the related CUDA kernel. 
temporary solution to support SYCL backend for stablediffusion.cpp https://github.com/leejet/stable-diffusion.cpp/issues/308

### CODE REVIEW

**Review by airMeng — APPROVED** [review]

**Comment by airMeng** [comment]
https://github.com/ggerganov/ggml/pull/900 please add t5 layernorm using this PR too

**Comment by zhentaoyu** [comment]
> [ggerganov/ggml#900](https://github.com/ggerganov/ggml/pull/900) please add t5 layernorm using this PR too

Will add them in another PR. Thanks.

**Comment by zhentaoyu** [comment]
rebased. @luoyu-intel please take a look. Thanks.

**Review by luoyu-intel — CHANGES_REQUESTED** [review]

File: ggml/src/ggml-sycl/tsembd.cpp [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    float freq = (float)sycl::native::exp(-(sycl::log((float)max_period)) * j / half);
+    float arg = timestep * freq;
+    embed_data[j] = sycl::cos(arg);
+    embed_data[j + half] = sycl::sin(arg);
+}
+
+static void timestep_embedding_f32_sycl(
+        const float * x, float * dst, const int ne00, const int nb1,
+        const int dim, const int max_period, const queue_ptr& stream) {
+    int half_ceil = (dim + 1) / 2;
```
[the remark this answers is not in the document]
Comment (zhentaoyu) [reply ggml/src/ggml-sycl/tsembd.cpp]: thanks. fixed

**Review by luoyu-intel — APPROVED** [review]

**Pull request merged**